### ISSUE TITLE
Ignore JSON files using .js file extension

### ISSUE THREAD

**Issue by Mogztter** [mozilla/addons-linter#3770]
### Describe the problem and steps to reproduce it:

[Asciidoctor Browser Extension](https://github.com/asciidoctor/asciidoctor-browser-extension) relies on MathJax (to render equations and formulas) which uses `.js` file extension on JSON files (see https://github.com/mathjax/MathJax/issues/2403).
Arguably, they should use `.json` file extension but they do not.

### What happened?

When running the linter on my extension, I get:

```
Code              Message                   Description                                                                                 File                                          Line   Column
JS_SYNTAX_ERROR   JavaScript syntax error   There is a JavaScript syntax error in your code; validation cannot continue on this file.   vendor/MathJax-3.0.5/sre/mathmaps/nemeth.js   2      30    
JS_SYNTAX_ERROR   JavaScript syntax error   There is a JavaScript syntax error in your code; validation cannot continue on this file.   vendor/MathJax-3.0.5/sre/mathmaps/fr.js       2      26    
JS_SYNTAX_ERROR   JavaScript syntax error   There is a JavaScript syntax error in your code; validation cannot continue on this file.   vendor/MathJax-3.0.5/sre/mathmaps/es.js       2      26    
JS_SYNTAX_ERROR   JavaScript syntax error   There is a JavaScript syntax error in your code; validation cannot continue on this file.   vendor/MathJax-3.0.5/sre/mathmaps/en.js       2      26 
```

So, basically I cannot use the latest version of MathJax because the linter is failing with an error.

### What did you expect to happen?

Would it be possible to ignore `.js` files that contain JSON or ignore a list of files?

**Comment by willdurand** [comment]
> Would it be possible to ignore `.js` files that contain JSON or ignore a list of files?

This seems unlikely but I'll wait for others to chime in.

**Comment by wagnerand** [comment]
I don't believe we should do that. JavaScript files need to be validated.

Adding custom detection of content or parts of it beyond the established ways does not sound like something we want to invest in or support. It sounds like a recipe for failure or maintenance hell at best.

Ignoring a list of files is not something we can do. At the latest when you submit the file to addons.mozilla.org, we'd need to include all files in validation.

**Comment by Mogztter** [comment]
Thanks for your reply!
I kind of agree with you but I was hoping for a solution/work around 😞 
I guess my only option is to repackage MathJax to either remove features associated with these files (speech-rule-engine) or update the file extension and the code accordingly.

If you can't think of any improvement in the linter that can help in these situations, I will close this issue.

Anyway, thanks again for looking into this issue 👍🏻
